### PR TITLE
DBFS Refactor & Fixes to support larger files

### PR DESCRIPTION
**Pull request by stikkireddy**
DBFS File had a few issues. This will be refactoring these issues:

1. Introduces two new fields: `source`, `content_b64_md5`
2. The new field `content_b64_md5` is required and it is used for determining differences and changes.
3. Updated documentation
4. Added unit testing and integration testing
5. Gives way forward for not clogging up state file with "content" data by using `source` field and also allows to go over the 4mb limit problem of GRPC limitations created by terraform. You can find the test here: `TestAccDatabricksDBFSFile_CreateVeryBigFiles`
6. Updated documentation to include new fields.

### CODE REVIEW

File: docs/resources/dbfs_file.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+The following arguments are supported:
+
+#### - `content`:
+> **(Optional)** The content of the file as a base64 encoded string.
+
+#### - `source`:
+> **(Optional)** The full absolute path to the file. Please use [pathexpand](https://www.terraform.io/docs/configuration/functions/pathexpand.html).
+
+#### - `content_b64_md5`:
+> **(Optional)** The checksum for the content please use the [md5](https://www.terraform.io/docs/configuration/functions/md5.html) and [filebase64](https://www.terraform.io/docs/configuration/functions/filebase64.html) functions in terraform to retrieve the checksum.
```
Comment (al-lac) [inline docs/resources/dbfs_file.md:45]: This should be set to `(Required)`

File: website/content/Resources/dbfs_file.md [hunk 2]
```diff
@@ -12,22 +12,44 @@ This is a resource that lets you create, get and delete files in DBFS (Databrick
...
 
 #### - `content`:
-> **(Required)** The content of the file as a base64 encoded string.
+> **(Optional)** The content of the file as a base64 encoded string.
+
+#### - `source`:
+> **(Optional)** The full absolute path to the file. Please use [pathexpand](https://www.terraform.io/docs/configuration/functions/pathexpand.html).
+
+#### - `content_b64_md5`:
+> **(Optional)** The checksum for the content please use the [md5](https://www.terraform.io/docs/configuration/functions/md5.html) and [filebase64](https://www.terraform.io/docs/configuration/functions/filebase64.html) functions in terraform to retrieve the checksum.
```
Comment (al-lac) [inline website/content/Resources/dbfs_file.md:52]: And here it should also be `(Required)`

File: databricks/resource_databricks_dbfs_file.go [hunk 3]
```diff
@@ -51,36 +70,56 @@ func resourceDBFSFile() *schema.Resource {
...
 	path := d.Get("path").(string)
-	content := d.Get("content").(string)
 	overwrite := d.Get("overwrite").(bool)
+	//checksum := d.Get("content_b64_md5").(string)
 	mkdirs := d.Get("mkdirs").(bool)
 
 	if mkdirs {
-		parentDir := filepath.Dir(path)
-		err := client.DBFS().Mkdirs(parentDir)
+		parentDir, err := GetParentDirPath(path)
```
Comment (nfx) [inline databricks/resource_databricks_dbfs_file.go:78]: (product code) this condition branch is about the side of the screen, so i'd suggest to move it to it's own dedicated function for redability

File: docs/resources/dbfs_file.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  validate_remote_file = true
+}
+```
+
+    
+## Argument Reference
+
+The following arguments are supported:
+
+#### - `content`:
```
Comment (nfx) [inline docs/resources/dbfs_file.md:38]: this doesn't look like registry documentation format :) please reformat

File: docs/resources/dbfs_file.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+> **(Optional)** The full absolute path to the file. Please use [pathexpand](https://www.terraform.io/docs/configuration/functions/pathexpand.html).
+
+#### - `content_b64_md5`:
+> **(Optional)** The checksum for the content please use the [md5](https://www.terraform.io/docs/configuration/functions/md5.html) and [filebase64](https://www.terraform.io/docs/configuration/functions/filebase64.html) functions in terraform to retrieve the checksum.
+
+#### - `path`:
+> **(Required)** The path of the file in which you wish to save.
+
+#### - `overwrite`:
+> **(Optional)** This is used to determine whether it should delete the
```
Comment (nfx) [inline docs/resources/dbfs_file.md:51]: please do not use new-line character within paragraphs

File: docs/resources/dbfs_file.md (hunk 1 above)
Comment (stikkireddy) [reply docs/resources/dbfs_file.md:45]: good catch thanks!

File: website/content/Resources/dbfs_file.md (hunk 2 above)
Comment (stikkireddy) [reply website/content/Resources/dbfs_file.md:52]: good catch thanks!

**Comment by stikkireddy** [comment]
@nfx @al-lac your change requests should have been fixed

**Pull request merged**